src/Elastic.CommonSchema/Serialization/EcsJsonConverterBase.cs

```diff
@@ -13,15 +13,16 @@ namespace Elastic.CommonSchema.Serialization
 	public abstract class EcsJsonConverterBase<T> : JsonConverter<T>
 	{
 		/// <summary></summary>
-		protected static bool ReadDateTime(ref Utf8JsonReader reader, ref DateTimeOffset? dateTime)
+		protected static bool ReadDateTime(ref Utf8JsonReader reader, ref DateTimeOffset? dateTime, JsonSerializerOptions options)
 		{
 			if (reader.TokenType == JsonTokenType.Null)
 			{
 				dateTime = null;
 				return true;
 			}
 
-			dateTime = EcsJsonConfiguration.DateTimeOffsetConverter.Read(ref reader, typeof(DateTimeOffset), EcsJsonConfiguration.SerializerOptions);
+			var converter = (JsonConverter<DateTimeOffset>)options.GetConverter(typeof(DateTimeOffset));
```

Comment:
It would be great if we can cache converters, or have a fast path for when options is the default instance for options?

Reply:
Now, at latest commit, we are back on caching the converter for the default options.
However, for custom options, I haven’t identified a unique identifier that could be used to store them in a dictionary-like structure.
Please, feel free to make any necessary modifications.

Reply:
Now, at latest commit, we are back on caching the converter for the default options.
However, for custom options, I haven’t identified a unique identifier that could be used to store them in a dictionary-like structure.
Please, feel free to make any modifications.